modules/Sources/Features/NHSendFlow/NHScan/NHScanStore.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  NHScanStore.swift
+//  
+//
+//  Created by Matthew Watt on 7/22/23.
+//
+
+import CaptureDevice
+import ComposableArchitecture
+import URIParser
+import Utils
+import ZcashLightClientKit
+
+public typealias NHScanStore = Store<NHScanReducer.State, NHScanReducer.Action>
+public typealias NHScanViewStore = ViewStore<NHScanReducer.State, NHScanReducer.Action>
+
+public struct NHScanReducer: ReducerProtocol {
+    private enum CancelId { case timer }
+    let networkType: NetworkType
+    
+    public struct State: Equatable {
+        public enum ScanStatus: Equatable {
+            case failed
+            case value(RedactableString)
+            case unknown
+        }
+        
+        public var scanStatus: ScanStatus = .unknown
+        
+        public var scannedValue: String? {
+            guard case let .value(scannedValue) = scanStatus else {
+                return nil
+            }
+            
+            return scannedValue.data
+        }
+        
+        public init() {}
+    }
+    
+    public enum Action: Equatable {
+        case backButtonTapped
+        case onAppear
+        case onDisappear
+        case found(RedactableString)
+        case scanFailed
+        case scan(RedactableString)
+    }
+    
+    public init(networkType: NetworkType) {
```

Comment:
[non-blocking] 

What's the code structure you are using? I see that the nested type and the parent type have their various code structures in different order.

For example `public struct State: Equatable ` has its initializer in the bottom of the declaration and the parent type has is between types and variables.

Reply:
Good call out. I have been undisciplined following the coding convention. I'll make sure to start following that.